### PR TITLE
Parse physical pixel sizes from CellSens .vsi datasets

### PR DESCRIPTION
**Pull request by melissalinkert**
Fixes http://trac.openmicroscopy.org.uk/ome/ticket/9805.  CellSens datasets (including test_images_good/cellsens) should now have PhysicalSizeX and PhysicalSizeY set for the largest resolution in each pyramid.  Smaller resolutions in the pyramid will not have physical sizes set, by design.

### CODE REVIEW

**Comment by melissalinkert** [comment]
Finally fixed the issues with reading .vsi files using Memoizer.  Removing the exclude label, as no more commits are planned and Travis is happy.

Note that this also fixes http://trac.openmicroscopy.org/ome/ticket/12433 now.  It would be best to spot check importing into OMERO the files from test_images_good, from_skyking/cellsens/olivier/small-tiles, and maybe one or two others.  Import should succeed, physical pixel sizes and exposure times should be present, and with the exception of test_images_good/anda pyramid generation should not occur.

File: components/formats-gpl/src/loci/formats/in/CellSensReader.java [hunk 1]
```diff
@@ -177,17 +427,18 @@ public boolean isSingleFile(String id) throws FormatException, IOException {
...
   /* @see loci.formats.IFormatReader#getOptimalTileWidth() */
   @Override
   public int getOptimalTileWidth() {
     FormatTools.assertId(currentId, true, 1);
-    if (getCoreIndex() < core.size() - ifds.size()) {
+    if (getCoreIndex() < core.size() - 1) {
       return tileX.get(getCoreIndex());
     }
-    int ifdIndex = ifds.size() - (core.size() - getCoreIndex());
+    int ifdIndex = 1 - (core.size() - getCoreIndex());
```
Comment (rleigh-dundee) [inline components/formats-gpl/src/loci/formats/in/CellSensReader.java]: Is this likely to result in a negative index?  It's also done below here.

**Comment by rleigh-dundee** [comment]
No import problems seen importing and viewing in OMERO.  Metadata looks good, and the expanded metadata coverage is a great improvement.  Looks good to merge.

File: components/formats-gpl/src/loci/formats/in/CellSensReader.java (hunk 1 above)
Comment (melissalinkert) [inline components/formats-gpl/src/loci/formats/in/CellSensReader.java]: No - the ```if``` clause on line 438 ensures that ```getCoreIndex()``` will be >= ```core.size() - 1``` if line 441 is executed.

**Comment by melissalinkert** [comment]
--no-rebase

**Pull request merged**